File: pinyougou-user-web/src/main/webapp/js/controller/userController.js
 //控制层 
app.controller('userController' ,function($scope,$controller   ,userService){	

	//注册
	$scope.reg = function () {
		if ($scope.entity.password != $scope.password){
			alert("两次密码不一致,请重新输入!");
			return;
		}
		userService.add($scope.entity,$scope.smsCode).success(
			function (response) {
				alert(response.message);
        })
    }

    //发送短信
	$scope.sendCode = function () {
        if ($scope.entity.phone == null || $scope.entity.phone == ""){
        	alert("手机号不能为空!");
        	return;
		}
		userService.sendCode($scope.entity.phone).success(
			function (response) {
				alert(response.message);
            }
		);
    }


});

File: pinyougou-user-web/src/main/java/com/pinyougou/user/controller/UserController.java
package com.pinyougou.user.controller;
import java.util.List;

import com.common.PhoneFormatCheckUtils;
import com.pinyougou.entity.PageResult;
import com.pinyougou.entity.Result;
import com.pinyougou.user.service.UserService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbUser;
/**
 * controller
 * @author Administrator
 *
 */
@RestController
@RequestMapping("/user")
public class UserController {

	@Reference
	private UserService userService;

	/**
	 * 增加
	 * @param user
	 * @return
	 */
	@RequestMapping("/add")
	public Result add(@RequestBody TbUser user,String smsCode){
		try {
			Boolean checkResult = userService.checkSmsCode(user.getPhone(), smsCode);
			if (checkResult == false){
				return new Result(false,"验证码错误");
			}
			userService.add(user);
			return new Result(true, "增加成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "增加失败");
		}
	}

	@RequestMapping("/sendCode")
	public Result sendCode(String phone){
		if (!PhoneFormatCheckUtils.isPhoneLegal(phone)){
			return new Result(false,"手机号错误,请重新输入!");
		}
		userService.createActiveCode(phone);
		return new Result(true,"验证短信已发送!");
	}
}
